misc-utils/lsfd.c

```diff
@@ -294,6 +294,27 @@ static const struct colinfo infos[] = {
 	[COL_TID]              = { "TID",
 				   5,   SCOLS_FL_RIGHT, SCOLS_JSON_NUMBER,
 				   N_("thread ID of the process opening the file") },
+	[COL_TIMERFD_CLOCKID]  = { "TIMERFD.CLOCKID",
+				   0,   SCOLS_FL_RIGHT, SCOLS_JSON_STRING,
+				   N_("clockid") },
+	[COL_TIMERFD_INTERVAL] = { "TIMERFD.INTERVAL",
+				   0,   SCOLS_FL_RIGHT, SCOLS_JSON_STRING,
+				   N_("interval") },
+	[COL_TIMERFD_INTERVAL_INT]={"TIMERFD.INTERVAL.INT",
+				   0,   SCOLS_FL_RIGHT, SCOLS_JSON_NUMBER,
+				   N_("integer part of TIMERFD.INTERVAL") },
+	[COL_TIMERFD_INTERVAL_NANOFRAC]={"TIMERFD.INTERVAL.NANOFRAC",
+				   0,   SCOLS_FL_RIGHT, SCOLS_JSON_NUMBER,
+				   N_("the nanosecond fraction part of TIMERFD.INTERVAL") },
+	[COL_TIMERFD_REMAINING] = { "TIMERFD.REMAINING",
+				   0,   SCOLS_FL_RIGHT, SCOLS_JSON_STRING,
```

Comment:
Couldn't this be `SCOLS_JSON_NUMBER` on its own, removing the need for the two derived columns?

Reply:
Thank you for reviewing.

The filtering engine supports only integers. So I don't want to introduce floating numbers to lsfd.

Reply:
Wouldn't it essentially force the users to implement their own floating point logic to use this properly?

`(TIMERFD.INTERVAL.INT >= 5 && TIMERFD.INTERVAL.NANOFRAC >= 5000000) && (TIMERFD.INTERVAL.INT < 6 && TIMERFD.INTERVAL.NANOFRAC < 5000000)`

seems very unwieldy.

What would be the issues to add float support to the filtering engine?

Reply:
> What would be the issues to add float support to the filtering engine?

My programming skills. When I designed the filtering engine, I assumed all numbers dealt with in lsfd are integers. With this simplification, I could implement the engine.

I'll look into the code of the filtering engine to support floating point column values.

Reply:
Now we can compare floating point numbers:

```
# ./lsfd -Q '(TIMERFD.REMAINING > 1.0) && (TIMERFD.REMAINING < 30.0)'
COMMAND       PID  USER ASSOC MODE    TYPE       SOURCE MNTID INODE NAME
gsd-color   17436   jet    11  rw- timerfd anon_inodefs    15  1060 clockid=realtime remaining=6.746271218
gsd-color 3370688 test2    15  rw- timerfd anon_inodefs    15  1060 clockid=realtime remaining=6.149405906
# ./lsfd -Q '(TIMERFD.REMAINING > 0.0) && (TIMERFD.REMAINING < 1.0)' 
COMMAND          PID        USER ASSOC MODE    TYPE       SOURCE MNTID INODE NAME
Xorg           17069         jet    26  rw- timerfd anon_inodefs    15  1060 clockid=monotonic remaining=0.061075544
systemd-oomd 2382701 systemd-oom     8  rw- timerfd anon_inodefs    15  1060 clockid=monotonic remaining=0.178126915
```